packages/shared-ux/chrome/navigation/types/index.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EuiSideNavItemType, IconType } from '@elastic/eui';
+import { Observable } from 'rxjs';
+import { BasePathService, NavigateToUrlFn, NavItemClickFn, RecentItem } from './internal';
+
+/**
+ * A list of services that are consumed by this component.
+ * @public
+ */
+export interface NavigationServices {
+  activeNavItemId?: string;
+  basePath: BasePathService;
+  loadingCount: number;
+  navIsOpen: boolean;
+  navigateToUrl: NavigateToUrlFn;
+  registerNavItemClick: NavItemClickFn;
+}
+
+/**
+ * An interface containing a collection of Kibana dependencies required to
+ * render this component
+ * @public
+ */
+export interface NavigationKibanaDependencies {
+  core: {
+    application: { navigateToUrl: NavigateToUrlFn };
+    chrome: {
+      getActiveNavItemId$: () => Observable<string | undefined>;
+      getProjectNavIsOpen$: () => Observable<boolean>;
+      recentlyAccessed: { get$: () => Observable<RecentItem[]> };
+      registerNavItemClick: (id: string) => void;
```

Comment:
This includes extensions that are currently not provided by the ChromeStart service. These extensions were used to work out active link highlighting, but we are planning to change the API so that active link highlighting and breadcrumbs can work based on something more structured in the link definition. I will remove this to make a clean PR that will work with minimal changes needed in ChromeStart.

Reply:
resolved in d1cd5512b8bbb4eaf29297a48fda046f474bb4d9